anndata/_core/sparse_dataset.py

```diff
@@ -474,6 +474,16 @@ def append(self, sparse_matrix: ss.spmatrix):
                 f"Matrices must have same format. Currently are "
                 f"{self.format!r} and {get_format(sparse_matrix)!r}"
             )
+        indptr_offset = self.group["indptr"][-1]
+        if indptr_offset.dtype == np.int32:
+            # This is difficult to test without requiring ~ 16GB of memory
+            new_nnz = int(indptr_offset) + int(sparse_matrix.indptr[-1])
```

Comment:
I don't think so, though I'm not sure I understand why it would be.

We could probably do `len(indptr)` which could be nicer from a performance perspective.

Reply:
I guess the reason I am bringing this up is two reasons.  Anecdotally, this seemed faster.  And second, I would assume that getting a shape on a view doesn't require reading into memory.

Also I think `len(indptr)` is not NNZ.  That's `indices`, and I found that that operation was also kinda slow.

From wikipedia:
_The arrays V and COL_INDEX are of length NNZ, and contain the non-zero values and the column indices of those values respectively
    COL_INDEX contains the column in which the corresponding entry V is located.
    The array ROW_INDEX is of length m + 1 and encodes the index in V and COL_INDEX where the given row starts. This is equivalent to ROW_INDEX[j] encoding the total number of nonzeros above row j. The last element is NNZ , i.e., the fictitious index in V immediately after the last valid index NNZ - 1.[[8]](https://en.wikipedia.org/wiki/Sparse_matrix#cite_note-8)_

Reply:
I will try a quick benchmark

Reply:
Ok, so this makes more sense.  My computer must have been having a day yesterday.  I don't think it really matters.  I was a little confused why reading one element of a view (as in the test) was taking so long, but I guess that's not an issue.